src/main/java/com/hazelcast/jdbc/HazelcastJdbcPreparedStatement.java

```
@@ -0,0 +1,370 @@
+    }
+
+    @Override
+    public void addBatch() throws SQLException {
+        checkClosed();
+        throw unsupportedBatch();
+    }
+
+    @Override
+    public void setCharacterStream(int parameterIndex, Reader reader, int length) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("Character Stream is not supported");
+    }
+
+    @Override
+    public void setRef(int parameterIndex, Ref x) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("Ref is not supported");
+    }
+
+    @Override
+    public void setBlob(int parameterIndex, Blob x) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("Blob is not supported");
+    }
+
+    @Override
+    public void setClob(int parameterIndex, Clob x) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("Clob is not supported");
+    }
+
+    @Override
+    public void setArray(int parameterIndex, Array x) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("Array is not supported");
+    }
+
+    @Override
+    public ResultSetMetaData getMetaData() throws SQLException {
+        checkClosed();
+        return null;
+    }
+
+    @Override
+    public void setDate(int parameterIndex, Date x, Calendar cal) throws SQLException {
+        checkClosed();
+    }
+
+    @Override
+    public void setTime(int parameterIndex, Time x, Calendar cal) throws SQLException {
+        checkClosed();
+    }
+
+    @Override
+    public void setTimestamp(int parameterIndex, Timestamp x, Calendar cal) throws SQLException {
+        checkClosed();
+    }
+
+    @Override
+    public void setNull(int parameterIndex, int sqlType, String typeName) throws SQLException {
+        checkClosed();
+        setParameter(parameterIndex, null);
+    }
+
+    @Override
+    public void setURL(int parameterIndex, URL x) throws SQLException {
+        checkClosed();
```

`setURL` ignores the passed value. Should we throw an exception here?